src/Roave/SecurityAdvisories/AdvisorySources/GetAdvisoriesFromGithubApi.php

```diff
@@ -85,8 +85,19 @@ public function __invoke(): Generator
     {
         foreach ($this->getAdvisories() as $item) {
             $versions = explode(',', $item['node']['vulnerableVersionRange']);
-            Assert::lessThanEq(count($versions), 2);
-            Assert::allStringNotEmpty($versions);
+
+            try {
+                iterable(
+                    int(),
+                    non_empty_string()
+                )->assert($versions);
+            } catch (AssertException $failure) {
+                throw new InvalidArgumentException($failure->getMessage(), (int) $failure->getCode(), $failure);
+            }
+
+            if (count($versions) > 2) {
+                throw new InvalidArgumentException('More than 2 version range delimiters found');
+            }
```

Comment:
you probably want `Type\shape([0 => Type\non_empty_string(), 1 => Type\non_empty_string()])->assert($versions);` here ( which means no need to check for the count )

Reply:
`$versions` can also contain 1 element here though, so I cannot expect key `1` to exist

Reply:
Type\union? :D